framework/bundles/org.jboss.tools.rsp.server/src/main/java/org/jboss/tools/rsp/server/model/ServerModel.java

```diff
@@ -484,4 +483,58 @@ private Attributes validateAttributes(Attributes ret, String serverType) {
 		}
 		return null;
 	}
+	
+	public IStatus addDeployable(IServer server, DeployableReference reference) {
+		IServerDelegate s = serverDelegates.get(server.getId());
+		if( s != null ) {
+			IStatus canAdd = s.canAddDeployable(reference);
+			if( canAdd.isOK()) {
+				s.getServerPublishModel().addDeployable(reference);
+				return Status.OK_STATUS;
+			} else {
+				return canAdd;
+			}
+		}
+		return new Status(IStatus.ERROR, ServerCoreActivator.BUNDLE_ID, 
+				"Server " + server.getId() + " not found.");
+	}
+
+	public IStatus removeDeployable(IServer server, DeployableReference reference) {
+		IServerDelegate s = serverDelegates.get(server.getId());
+		if( s != null ) {
+			IStatus canRemove = s.canRemoveDeployable(reference);
+			if( canRemove.isOK()) {
+				s.getServerPublishModel().removeDeployable(reference);
+				return Status.OK_STATUS;
+			} else {
+				return canRemove;
+			}
+		}
+		return new Status(IStatus.ERROR, ServerCoreActivator.BUNDLE_ID, 
+				"Server " + server.getId() + " not found.");
+	}
+
+	public List<DeployableState> getDeployables(IServer server) {
+		IServerDelegate s = serverDelegates.get(server.getId());
+		if( s != null ) {
+			return s.getServerPublishModel().getDeployables();
+		}
+		return new ArrayList<DeployableState>();
+	}
+
+	@Override
+	public IStatus publish(IServer server, int kind) throws CoreException {
+		IServerDelegate s = serverDelegates.get(server.getId());
+		if( s != null ) {
+			IStatus canPublish = s.canPublish();
+			if( canPublish != null && canPublish.isOK()) {
+				return s.publish(kind);
+			} else {
+				return new Status(IStatus.ERROR, ServerCoreActivator.BUNDLE_ID, 
```

Comment:
return canPublish instead like you do in the code that adds/removes deployables?